common/src/main/java/com/google/udmi/util/JsonUtil.java

```diff
@@ -486,4 +489,121 @@ public static void writeFile(Object theThing, File file) {
       throw new RuntimeException("While writing " + file.getAbsolutePath(), e);
     }
   }
+
+  public static Map<String, Object> flattenNestedMap(Map<String, Object> map, String separator) {
+    Map<String, Object> flattenedMap = new TreeMap<>();
+    flatten(map, "", flattenedMap, separator);
+    return flattenedMap;
+  }
+
+  @SuppressWarnings("unchecked")
+  private static void flatten(Map<String, Object> currentMap, String currentKey,
+      Map<String, Object> flattenedMap, String separator) {
+    for (Map.Entry<String, Object> entry : currentMap.entrySet()) {
+      String key = entry.getKey();
+      Object value = entry.getValue();
+      String newKey = currentKey.isEmpty() ? key : currentKey + separator + key;
+
+      if (value instanceof Map) {
+        flatten((Map<String, Object>) value, newKey, flattenedMap, separator);
+      } else {
+        flattenedMap.put(newKey, value);
+      }
+    }
+  }
+
+  public static JsonNode nestFlattenedJson(Map<String, String> flattenedJsonMap) {
+    ObjectMapper mapper = new ObjectMapper();
+    ObjectNode rootNode = mapper.createObjectNode();
+
+    for (Map.Entry<String, String> entry : flattenedJsonMap.entrySet()) {
+      String key = entry.getKey();
+      String value = entry.getValue();
+      String[] parts = key.split("\\.");
+      nest(rootNode, parts, value, 0, mapper);
+    }
+
+    return rootNode;
+  }
+
+  private static void nest(JsonNode currentNode, String[] parts, String value,
+      int index, ObjectMapper mapper) {
+    if (index == parts.length - 1) {
+      if (currentNode instanceof ObjectNode) {
+        ((ObjectNode) currentNode).put(parts[index], value);
+      } else if (currentNode instanceof ArrayNode) {
+        try {
+          int arrayIndex = Integer.parseInt(parts[index]);
+          ((ArrayNode) currentNode).set(arrayIndex, mapper.getNodeFactory().textNode(value));
+        } catch (NumberFormatException e) {
+          System.err.println("Warning: Cannot set array value at non-numeric index: " + parts[index]);
```

Comment:
shouldn't silently squelch this error here -- likely should catch and then rethrow as a RuntimeException